evalml/tests/automl_tests/test_automl_utils.py

```diff
@@ -93,6 +98,20 @@ def test_make_data_splitter_default(problem_type, large_data):
         assert data_splitter.test_size == _LARGE_DATA_PERCENT_VALIDATION
         return
 
+    if data_size == "small" and problem_type in [
+        ProblemTypes.REGRESSION,
+        ProblemTypes.BINARY,
+        ProblemTypes.MULTICLASS,
+    ]:
+        assert isinstance(
+            data_splitter,
+            KFold if problem_type == ProblemTypes.REGRESSION else StratifiedKFold,
+        )
+        assert data_splitter.n_splits == _SMALL_DATA_CV_FOLDS
+        assert data_splitter.shuffle
+        assert data_splitter.random_state == 0
+        return
+
```

Comment:
Perhaps this branch should merge with the top branch as all of them branch off the same group of problem types?

```
    if problem_type in [
        ProblemTypes.REGRESSION,
        ProblemTypes.BINARY,
        ProblemTypes.MULTICLASS,
    ]:
        if data_size == "large":
            assert isinstance(data_splitter, TrainingValidationSplit)
            assert data_splitter.stratify is None
            assert data_splitter.random_seed == 0
            assert data_splitter.shuffle
            assert data_splitter.test_size == _LARGE_DATA_PERCENT_VALIDATION
        elif data_size == "small" 
            assert isinstance(
                data_splitter,
                KFold if problem_type == ProblemTypes.REGRESSION else StratifiedKFold,
            )
            assert data_splitter.n_splits == _SMALL_DATA_CV_FOLDS
            assert data_splitter.random_state == 0
            assert data_splitter.shuffle
```